Tests/AsyncKitTests/ConnectionPoolTests.swift

```diff
@@ -384,26 +390,23 @@ private final class FooDatabase: ConnectionPoolSource {
     init() {
         self.connectionsCreated = .init(0)
     }
-    
+
     func makeConnection(logger: Logger, on eventLoop: EventLoop) -> EventLoopFuture<FooConnection> {
         let conn = FooConnection(on: eventLoop)
         self.connectionsCreated.wrappingIncrement(by: 1, ordering: .relaxed)
         return conn.eventLoop.makeSucceededFuture(conn)
     }
 }
 
-private final class FooConnection: ConnectionPoolItem {
-    var lastUsed: Date
-    
+private final class FooConnection: ConnectionPoolItem, AtomicReference {
```

Comment:
vapor has it conformed to `AtomicReference`, didn't investigate why or if it's any useful anywhere anyway.